boto/dynamodb2/results.py

```diff
@@ -180,6 +180,9 @@ def fetch_more(self):
         kwargs['keys'] = self._keys_left[:self._max_batch_get]
         self._keys_left = self._keys_left[self._max_batch_get:]
 
+        if len(self._keys_left) <= 0:
```

Comment:
Is it necessary to invoke the_callable if len(_keys_left) <= 0? It seems like you should be able to return before then.

If _keys_left is [], then it seems like unprocessed_keys should always be [] later on, which would allow you to avoid the call and remove lines 199-200.